src/main/java/org/italiangrid/storm/webdav/server/servlet/PreAuthenticatedFilter.java

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: 2014 Istituto Nazionale di Fisica Nucleare
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.italiangrid.storm.webdav.server.servlet;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import org.italiangrid.storm.webdav.authn.PrincipalHelper;
+import org.italiangrid.storm.webdav.server.tracing.RequestIdHolder;
+import org.slf4j.MDC;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+// ForwardedHeaderFilter does not parse the "by" part of the Forwarded header
+// This wraps the request so the local address/port are correct
+// https://github.com/spring-projects/spring-framework/issues/34654
+public class PreAuthenticatedFilter extends OncePerRequestFilter {
+
+  public static final String REQUEST_ID_ATTRIBUTE_NAME = "storm.requestId";
+  public static final String REMOTE_USER_ATTR_NAME = "storm.remoteUser";
+  public static final String REMOTE_ADDR_NAME = "storm.remoteAddr";
+
+  private final boolean nginxEnabled;
+  private final PrincipalHelper helper;
+
+  public PreAuthenticatedFilter(boolean nginxEnabled, PrincipalHelper principalHelper) {
+    this.nginxEnabled = nginxEnabled;
+    this.helper = principalHelper;
+  }
+
+  @Override
+  protected void doFilterInternal(
+      HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+      throws IOException, ServletException {
+    String requestId;
+    if (nginxEnabled) {
+      requestId = request.getHeader("X-nginx-request_id");
+    } else {
+      RequestIdHolder.setRandomId();
+      requestId = RequestIdHolder.getRequestId();
+    }
```

Comment:
```suggestion
    if (nginxEnabled) {
      requestId = request.getHeader("x-request-id");
    }
    if (requestId.isEmpty()) {
      RequestIdHolder.setRandomId();
      requestId = RequestIdHolder.getRequestId();
    }
```
Per proteggersi nel caso Nginx non passi l'header. Ho usato il nome dell'header usato anche in storm-tape (vedi altro commento).

Reply:
Aggiungo che non capisco l'uso di `RequestIdHolder` e che bisogno ci sia di averlo thread-local, visto che poi il requestId calcolato viene messo in MDC

Reply:
Added fallback in case the `X-request-id` header is not present.